Fix "Lint Code Base" shellcheck workflow errors

Fix the failing checks from the "Lint Code Base" shellcheck workflow step:
```
File:[./entrypoint.sh]
ERROR! Found errors in [shellcheck] linter!
ERROR:[
In ./entrypoint.sh line 3:
git clone --quiet https://github.com/$REPO &> /dev/null
                                     ^---^ SC2086: Double quote to prevent globbing and word splitting.
                                           ^----------^ SC2039: In POSIX sh, &> is undefined.

Did you mean: 
git clone --quiet https://github.com/"$REPO" &> /dev/null


In ./entrypoint.sh line 12:
echo $changelog
     ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
echo "$changelog"


In ./entrypoint.sh line 14:
changelog="${changelog//'%'/'%25'}"
           ^---------------------^ SC2039: In POSIX sh, string replacement is undefined.


In ./entrypoint.sh line 15:
changelog="${changelog//$'\n'/'%0A'}"
           ^-----------------------^ SC2039: In POSIX sh, string replacement is undefined.
                        ^---^ SC2039: In POSIX sh, $'..' is undefined.


In ./entrypoint.sh line 16:
changelog="${changelog//$'\r'/'%0D'}"
           ^-----------------------^ SC2039: In POSIX sh, string replacement is undefined.
                        ^---^ SC2039: In POSIX sh, $'..' is undefined.

For more information:
  https://www.shellcheck.net/wiki/SC2039 -- In POSIX sh, $'..' is undefined.
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...]
```
https://github.com/ScottBrenner/generate-changelog-action/runs/1129086107?check_suite_focus=true#step:4:71

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -2,7 +2,7 @@
 
 git clone --quiet https://github.com/"$REPO"
 
-if [ "$REPO" == "ScottBrenner/generate-changelog-action" ]; then
+if [ "$REPO" = "ScottBrenner/generate-changelog-action" ]; then
   cd generate-changelog-action
 fi
 

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -3,7 +3,7 @@
 git clone --quiet https://github.com/"$REPO"
 
 if [ "$REPO" = "ScottBrenner/generate-changelog-action" ]; then
-  cd generate-changelog-action
+  cd generate-changelog-action || exit
 fi
 
 if [ "$1" ] && [ "$1" != "package.json" ]; then

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -1,6 +1,5 @@
 #!/bin/sh -l
-# shellcheck disable=SC3060
-# shellcheck disable=SC3003
+# shellcheck disable=SC2039
 
 git clone --quiet https://github.com/"$REPO"
 

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -17,6 +17,6 @@ changelog="${changelog//'%'/'%25'}"
 changelog="${changelog//$'\n'/'%0A'}"
 changelog="${changelog//$'\r'/'%0D'}"
 
-echo $changelog
+echo "$changelog"
 
 echo "::set-output name=changelog::$changelog"

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -1,6 +1,6 @@
 #!/bin/sh -l
 
-git clone --quiet https://github.com/$REPO &> /dev/null
+git clone --quiet https://github.com/"$REPO" &> /dev/null
 
 if [ "$REPO" == "ScottBrenner/generate-changelog-action" ]; then
   cd generate-changelog-action

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -1,5 +1,6 @@
 #!/bin/sh -l
-# shellcheck disable=SC3000-SC4000
+# shellcheck disable=SC3060
+# shellcheck disable=SC3003
 
 git clone --quiet https://github.com/"$REPO"
 

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -1,6 +1,6 @@
 #!/bin/sh -l
 
-git clone --quiet https://github.com/"$REPO" &> /dev/null
+git clone --quiet https://github.com/"$REPO"
 
 if [ "$REPO" == "ScottBrenner/generate-changelog-action" ]; then
   cd generate-changelog-action

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -1,4 +1,5 @@
 #!/bin/sh -l
+# shellcheck disable=SC3000-SC4000
 
 git clone --quiet https://github.com/"$REPO"
 

